helix-core/src/main/java/org/apache/helix/monitoring/metrics/model/CountMetric.java

```diff
@@ -19,23 +19,46 @@
  * under the License.
  */
 
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.DynamicMetric;
 import org.apache.helix.monitoring.mbeans.dynamicMBeans.SimpleDynamicMetric;
 
 /**
  * Represents a count metric and defines methods to help with calculation. A count metric gives a
  * gauge value of a certain property.
  */
-public abstract class CountMetric<V> extends SimpleDynamicMetric<V> implements Metric {
-  protected V _count;
+public abstract class CountMetric extends SimpleDynamicMetric<Long> implements Metric {
```

Comment:
I recall having this discussion with you. Should we specify the exact type in the subclass of `CountMetric` in their constructors? That allows other subclasses of CountMetric to specify their own SimpleDynamicMetric.

Reply:
Note that Metric interface has defined long as the return value of getLastEmittedMetricValue(). So I don't think we have a chance of using a different type in the sub-class.

Reply:
@pkuwm Did you have a discussion on this with @jiajunwang ? I think you wanted to use a different type?

Reply:
Discussed with @pkuwm, that will not be a counter. If we want to report a float for usage, we would have to create a GuageMetric. As for the count, it will be Long, since we are counting one by one.